src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -1043,37 +1059,106 @@ export class BundleDataClient {
                 this.getRelayHashFromEvent(matchedDeposit) === relayDataHash,
                 "Deposit relay hashes should match."
               );
+              v3RelayHashes[relayDataHash].deposit = matchedDeposit;
```

Comment:
Moved this higher up since we're now exiting below with the `_canCreateSlowFillLeaf` check. There is some change here where a deposit is not involved with a lite chain, gets saved into the `v3Relayhashes` dictionary before the code exits because the deposit is a token swap or the deposit has expired. The new behavior is to save the deposit into `v3Relayhashes` dictionary and then exit if any of those checks fail. I don't see any issues with this small change